ConservationInternational/resilienceatlas
# RA2-216: Adding timeline attributes to Layer (model, API, backoffice)

Adding all required attributes to Layer model which are mandatory for timeline (chronological data) to work. 

```ruby
add_column :layers, :timeline, :boolean, default: false
add_column :layers, :timeline_overlap, :string
add_column :layers, :timeline_steps, :date, array: true, default: []
add_column :layers, :timeline_start_date, :date
add_column :layers, :timeline_end_date, :date
add_column :layers, :timeline_default_date, :date
add_column :layers, :timeline_period, :string
add_column :layers, :timeline_format, :string, default: "%Y-%m-%d"
```

All these attributes can be managed via Backoffice and are available through API endpoint. Only interesting part is `timeline_steps` which is array of dates so this one can be defined at Admin as date in format `%Y-%m-%d` and individual dates are divides by `,`.

## Testing instructions

Open Admin and try to edit timeline information of some Layer. Check that information gets saved and data are visible at Layer detail at backoffice. Call `api/layers` endpoint and check that layers data contains timeline attributes. Please after you finish testing this, disable timeline again ;)

## Tracking

https://vizzuality.atlassian.net/browse/RA2-216
